[3.2] docs(pager): Revamped Pager Overview (#828)
telerik/blazor-docs: components/pager/overview.md

The Pager provides the UI for the user to change the page. To the developer, it provides the page index so you can render only the relevant data portion and an [event]({%slug pager-events%}) that you can use to implement [load on demand](#load-on-demand).

## Creating Blazor Pager

1. Add the `TelerikPager` tag to a Razor file.

1. Set the `Total` property to the number of items in the data source.

> 1. Set the `PageSize` and `Page` (one-way or two-way binding) parameters values to extract and render the desired subset of data. The page indexes are 1-based.

Reviewer: Why two parameters in a single step?

>caption Use the TelerikPager to paginate your own data and content.

````CSHTML
<TelerikPager Total="@Games.Count" PageSize="@PageSize" @bind-Page="@Page"></TelerikPager>

@{
    // take and render the relevant data portion based on the pager info
    var pageData = Games.Skip((Page - 1) * PageSize).Take(PageSize).ToList();

    @foreach (Game game in pageData)
    {
        <div style="display: inline-block;border: solid;padding: 10px;margin: 10px">
            <h5>@game.GameName</h5>
            <h6>@game.GameId</h6>
        </div>

>     }

Reviewer: Why h5 and h6?

```
    @foreach (Game game in pageData)
    {
        <div class="k-panel" style="margin: 1em; padding: 1em;">
            @game.GameName
        </div>
    }
```

}

@code {
    public int PageSize { get; set; } = 3;
    public int Page { get; set; } = 1;

    public List<Game> Games { get; set; }

    protected override void OnInitialized()
    {
        Games = new List<Game>();
        for (int i = 1; i < 20; i++)
        {
            Games.Add(new Game()
                {
                    GameName = $"Game {i}",

>                     GameId = i,

Reviewer: We normally put the ID first.

| `PageSize` | `int` | The number of items to display on a page. |
| `PageSizes` | `List<int?>` | Allows users to change the page size via a DropDownList. The attribute configures the DropDownList options. A `null` item in the `PageSizes` `List` will render an "All" option. By default, the Pager DropDownList is not displayed. You can also set `PageSizes` to `null` programmatically to remove the DropDownList at any time. |
| `InputType` | `PagerInputType` enum <br /> (`Buttons`) | Determines if the pager will show numeric buttons to go to a specific page, or a textbox to type the page index. The arrow buttons are always visible. The `PagerInputType` enum accepts values `Buttons` (default) or `Input`. When `Input` is used, the page index will change when the textbox is blurred, or when the user hits Enter. This is to avoid unintentional data requests. |

> | `Class` | `string` | The CSS class that will be rendered on the main wrapping element of the Pager. |

Reviewer: Which element exactly?

The Pager provides the UI for the user to change the page. To the developer, it provides the page index so you can render only the relevant data portion and an [event]({%slug pager-events%}) that you can use to implement [load on demand](#load-on-demand).

## Creating Blazor Pager

1. Add the `TelerikPager` tag to a Razor file.

2. Set the `Total` property to the number of items in the data source.

> 3. Set the `PageSize` parameter value to extract and render the desired subset of data.

Reviewer: ```suggestion
3. Set the `PageSize` parameter to determine the desired subset of data. It supports two-way binding.
```

    {
        Games = new List<Game>();
        for (int i = 1; i < 20; i++)
        {
            Games.Add(new Game()
                {
                    GameId = i,
                    GameName = $"Game {i}"
                });
        }
    }

    public class Game
    {
        public int GameId { get; set; }
        public string GameName { get; set; }
    }
}
````

## Events

The Blazor Pager exposes PageChanged and PageSizeChanged events that you can handle and further customize its behavior. [Read more about the Blazor Pager events]({%slug pager-events%}).

## Pager Parameters

The Blazor Pager provides various parameters that allow you to configure the component:

<style>
    article style + table {
        table-layout: auto;
        word-break: normal;
    }
</style>
| Parameter | Type and Default Value | Description |
| ----------- | ----------- | ----------- |
| `ButtonCount` | `int` | The maximum number of page buttons that will be visible. To take effect, `ButtonCount` must be smaller than the page count (`ButtonCount < Total / PageSize`). |
| `Class` | `string` | Renders a custom CSS class to the `<div class="k-pager-wrap">` element. |

> | `Page` | `int` | Represents the current page of the pager. The first page has an index of `1`. Supports two-way data binding. If no value is provided, the parameter will default to the first page (1), but you should always use this parameter value in order to successfully use the component. If you don't use two-way binding and you don't update the value of the parameter after the user action, the pager UI will not reflect the change and will revert to the previous value (page index). |

Reviewer: ```suggestion
| `Page` | `int` | Represents the current page of the pager. The first page has an index of `1`. Supports two-way binding. If no value is provided, the parameter will default to the first page (1), but you should always use this parameter value in order to successfully use the component. If you don't use two-way binding and you don't update the value of the parameter after the user action, the pager UI will not reflect the change and will revert to the previous value (page index). |
```

> | `PageSize` | `int` | The number of items to display on a page. |

Reviewer: ```suggestion
| `PageSize` | `int` | The number of items to display on a page. Supports two-way binding. |
```
